src/middleware.ts

```diff
@@ -136,9 +136,9 @@ export const combine = <
     {},
     initialState,
     create(
-      set as SetState<PrimaryState>,
+      (set as unknown) as SetState<PrimaryState>,
```

Comment:
This looks unfortunate, but can be accepted.

Reply:
Agreed. I haven't found a way around it :(